rust/src/settings.rs

```diff
@@ -436,6 +437,42 @@ pub struct ModelSettings {
     pub size: usize,
 }
 
+#[derive(Debug, Deserialize)]
+/// Metrics settings.
+pub struct MetricsSettings {
+    /// The URL address of the metrics store.
+    ///
+    /// # Examples
+    ///
+    /// **TOML**
+    /// ```text
+    /// [metrics]
+    /// store_url = "http://localhost:8086"
+    /// ```
+    ///
+    /// **Environment variable**
+    /// ```text
+    /// XAYNET_API__STORE_URL=http://localhost:8086
+    /// ```
+    pub store_url: String,
```

Comment:
Do you know if it's easy to check if the value is a valid URL?

For example you can use the type `std::net::SocketAddr` for ip addresses. 
When the configuration file is parsed, `SocketAddr::FromStr` is called and it will automatically fail if it is not a valid IP address. We used that behaviour for `ApiSettings::bind_address`. Maybe there is something similar for urls in the std lib.

However, if it requires an additional library, I'm not sure it's worth it.

Reply:
We can easily check whether a URL is valid, whether is contains a hostname, what the scheme is, etc. But we cannot check that the domain is valid in that it resolves to an IP address without making a network call, and I think we should refrain from doing network calls when parsing a config file, because they can hang.

Reply:
one thing we can certainly do I think is `#[validate(url)]`. I'll add that now...

Reply:
Right, I just thought of checking that scheme is valid. 
But maybe it's not that important for now (just an idea I had in mind).